Allow running changed packages on target branches (#2)
circletron/circletron: readme.md

## Details

It is useful to set up branch protection rules to prevent code from being merged when a CI job does not pass. When jobs are omitted then the PR will never be mergeable since the job will remain in a `pending` state. For this reason `circletron` will never omit a job that was determined not to be run, instead the job will be replaced wit a simple job that echos "Job is not required" and return a success exit status.

## Configuration Properties

> Circletron configuration can be provided in a `.circle/circletron.yml` file. You can specify target branches using `targetBranches`:

Reviewer: It might be better to explain the `targetBranches` configuration after explaining what target branches are rather than before?

Reviewer: Also there is already some documentation on line 31-38 that explains what this configuration is, I agree it's nice to group the configuration properties into a section, but would want to avoid duplicating information.

```
# this is the default value
targetBranches: ^(release/|main$|master$|develop$)
```

> On target branches, Circletron can be configured to only run workflows in the packages that have changed since the last successful build on that branch. This feature interacts with the Circle API v2, so requires an access token to be provided, via the `CIRCLE_TOKEN` environment variable. You can turn this on using `runOnlyChangedOnTargetBranches`:

Reviewer 2: ```suggestion
On target branches, Circletron can be configured to only run workflows in the packages that have changed since the last successful build on that branch. This feature interacts with the Circle API v2, so requires an access token to be provided, via the `CIRCLE_TOKEN` environment variable. Turn on this feature by enabling `runOnlyChangedOnTargetBranches`:
```

Reviewer: Ah yeah @sxwei123 has a good point here, the documentation is currently in impersonal mood, it'd be nice to stay consistent on that. Impersonal mood usually ends up terser and I think it sounds a little more professional.
